pkg/controller/wip/controller_test.go

```diff
@@ -136,6 +138,81 @@ func TestReconcileBroker(t *testing.T) {
 
 }
 
+func TestReconcileBrokerDelete(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, _, _, _, testController, sharedInformers, stopCh := newTestController(t)
+	defer close(stopCh)
+
+	testServiceClass := getTestServiceClass()
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(testServiceClass)
+
+	broker := getTestBroker()
+	broker.DeletionTimestamp = &metav1.Time{}
+	broker.Finalizers = []string{"kubernetes"}
+
+	testController.reconcileBroker(broker)
+
+	// Verify no core kube actions occurred
+	kubeActions := fakeKubeClient.Actions()
+	if e, a := 0, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	// The three actions should be:
+	// 0. Deleting the associated ServiceClass
+	// 1. Updating the ready condition
+	// 2. Removing the finalizer
+	if e, a := 3, len(actions); e != a {
+		t.Logf("%+v\n", actions)
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+
+	deleteAction := actions[0].(core.DeleteActionImpl)
+	if e, a := "delete", deleteAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[0]; expected %v, got %v", e, a)
+	}
+
+	if e, a := testServiceClass.Name, deleteAction.Name; e != a {
+		t.Fatalf("Unexpected name of serviceclass: expected %v, got %v", e, a)
+	}
+
+	updateAction := actions[1].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[1]; expected %v, got %v", e, a)
+	}
+
+	updatedBroker := updateAction.GetObject().(*v1alpha1.Broker)
```

Comment:
`updatedBroker` will be `nil` if the type assertion failed. the test will still fail, but the error message won't be as clear. not a big deal; I'm ok with this :+1: